evm/src/witness/memory.rs

```diff
@@ -69,6 +71,14 @@ impl MemoryAddress {
         })
     }
 
+    pub(crate) fn new_bundle(addr: U256) -> Result<Self, ProgramError> {
```

Comment:
Docs perhaps?

Reply:
Why do we need to convert the limbs into `U256` and then run the checks in `new_u256s()` and then convert to `usize`? Isn't there a way to split `addr` into the three `usize` we need and check `bits()`/`Segment::COUNT` on them directly? Save a round trip?